### PR TITLE
[Deployment] Secure kubelet public api

### PR DESCRIPTION
**Pull request by abuccts**
Secure kubelet public api, add authentication for https port 10250 and close http port 10255.

Please refer to https://github.com/kubernetes/kubernetes/issues/7965, https://github.com/kubernetes/kubernetes/pull/59666 for reasons and https://github.com/docker/for-linux/issues/324, [Backdooring through kubelet](https://medium.com/handy-tech/analysis-of-a-kubernetes-hack-backdooring-through-kubelet-823be5c3d67c) for hacking examples.

### CODE REVIEW

**Comment by hao1939** [comment]
@ydye , this PR breaks the deployment process, could you take a look?

File: pai-management/k8sPaiLibrary/template/certs.sh.template [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+authorityKeyIdentifier=keyid,issuer:always
+basicConstraints=CA:FALSE
+keyUsage=keyEncipherment,dataEncipherment
+extendedKeyUsage=serverAuth,clientAuth
+subjectAltName=@alt_names
+EOL
+
+
+# Authentication for apiserver -> kubelet
+openssl genrsa -out ${certspath}/kubelet.ca.key 4096
```
Comment (fanyangCS) [inline pai-management/k8sPaiLibrary/template/certs.sh.template]: >openssl [](start = 0, length = 7)

do you assume the host OS always has openssl?

Comment (YitongFeng) [reply pai-management/k8sPaiLibrary/template/certs.sh.template]: https://github.com/Microsoft/pai/blob/c01981c68963d960beec37b9e558ad98de853159/pai-management/k8sPaiLibrary/maintaintool/docker-ce-install.sh#L43

When install docker, we installed ca-certificates which depends on openssl, and will install openssl first.

**Comment by xudifsd** [comment]
should remove kubelet health metrics from watchdog after this PR is merged.

**Comment by fanyangCS** [comment]
close this PR and address this issue in other PRs.

**Pull request closed without merging**